delivery_auto_refresh/models/sale_order.py

```diff
@@ -29,14 +57,16 @@ def create(self, vals):
 
     def write(self, vals):
         """Create or refresh delivery line after saving."""
-        res = super(SaleOrder, self).write(vals)
-        for order in self:
-            delivery_line = order.order_line.filtered("is_delivery")
-            if len(delivery_line) > 1:
-                continue
-            order.with_context(
-                delivery_discount=delivery_line.discount,
-            )._auto_refresh_delivery()
+        res = super().write(vals)
+        if (
+            self._get_param_auto_add_delivery_line()
+            and "auto_refresh_delivery" not in self.env.context
```

Comment:
This would be more resilient:
```suggestion
            and not self.env.context.get("auto_refresh_delivery")
```